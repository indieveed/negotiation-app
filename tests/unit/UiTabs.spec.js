import { shallowMount } from '@vue/test-utils'
import UiTabs from '@/components/ui/tabs.vue'

const items = [{
  text: 'Employer',
  id: 0,
  active: true
}, {
  text: 'Employee',
  id: 1,
  active: false
}]


describe('ui/tabs.vue', () => {
  it('renders the component', () => {
    const wrapper = shallowMount(UiTabs, {
      propsData: {
        items
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders the same amount of labels as tabs passed', () => {
    const wrapper = shallowMount(UiTabs, {
      propsData: {
        items
      }
    })
    expect(wrapper.findAll('.tabs__label').length).toEqual(items.length)
  })
  it('renders the same amount of panels as tabs passed', () => {
    const wrapper = shallowMount(UiTabs, {
      propsData: {
        items
      }
    })
    expect(wrapper.findAll('.tabs__panel').length).toEqual(items.length)
  })
  it('renders the scoped slots', () => {
    const wrapper = shallowMount(UiTabs, {
      propsData: {
        items
      },
      scopedSlots: {
        panel: `<div slot-scope="{item}">{{item.id}}</div>`
      }
    })
    const panels = wrapper.findAll('.tabs__panel')
    for (let index = 0; index < panels.length; index++) {
      expect(panels.at(index).text()).toEqual(items[index].id.toString())
    }
  })
  it('labels \' text is passed through props', () => {
    const wrapper = shallowMount(UiTabs, {
      propsData: {
        items
      }
    })
    const labels = wrapper.findAll('.tabs__label')
    for (let index = 0; index < labels.length; index++) {
      expect(labels.at(index).text()).toEqual(items[index].text)
    }
  })
})
