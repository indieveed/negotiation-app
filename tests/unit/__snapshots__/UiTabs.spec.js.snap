// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/tabs.vue renders the component 1`] = `
<div class="tabs">
  <div role="tablist" class="tabs__header"><input id="tabs-1-0" name="tabs-1" type="radio" role="tab" class="tabs__input" value="tabs-1-0"> <label for="tabs-1-0" class="tabs__label">Employer</label><input id="tabs-1-1" name="tabs-1" type="radio" role="tab" class="tabs__input" value="tabs-1-1"> <label for="tabs-1-1" class="tabs__label">Employee</label></div>
  <div class="tabs__body">
    <div name="tabs-1" role="tabpanel" tabindex="0" aria-labelledby="tabs-1-0" class="tabs__panel is-current"></div>
    <div name="tabs-1" role="tabpanel" tabindex="0" aria-labelledby="tabs-1-1" class="tabs__panel"></div>
  </div>
</div>
`;
